spikeinterface/toolkit/postprocessing/correlograms.py

```diff
@@ -1,6 +1,71 @@
 import numpy as np
 
 
+try:
+    import numba
+    HAVE_NUMBA = True
+except ModuleNotFoundError as err:
+    HAVE_NUMBA = False
+
+def compute_autocorrelogram_from_spiketrain(spike_train, max_time, bin_size, sampling_f):
+    """
+    Compute the auto-correlogram from a given spike train.
+
+    This implementation only works if you have numba installed, to accelerate the
+    computation time.
+
+    Parameters
+    ----------
+    spike_train: np.ndarray
+        The ordered spike train to compute the auto-correlogram.
+    max_time: float
+        Compute the auto-correlogram between -max_time and +max_time (in ms).
+    bin_size: float
+        Size of a bin (in ms).
+    sampling_f: float:
+        Sampling rate/frequency (in Hz).
+
+    Returns
+    -------
+    tuple (auto_corr, bins)
+    auto_corr: np.ndarray[int64]
+        The computed auto-correlogram.
+    bins: np.ndarray[float64]
+        The bins for the auto-correlogram.
+    """
+    if not HAVE_NUMBA:
+        print("Error: numba is not installed.")
+        print("compute_autocorrelogram_from_spiketrain cannot run without numba.")
+        return 0
+
+    return _compute_autocorr_numba(spike_train.astype(np.int64), max_time, bin_size, sampling_f)
+
+if HAVE_NUMBA:
+    @numba.jit((numba.int64[::1], numba.float32, numba.float32, numba.float32), nopython=True,
+                nogil=True, cache=True)
+    def _compute_autocorr_numba(spike_train, max_time, bin_size, sampling_f):
+        bin_size_c = int(round(bin_size * 1e-3 * sampling_f))
+        max_time_c = int(round(max_time * 1e-3 * sampling_f))
+        max_time -= max_time / bin_size
+        n_bins = 2 * int(max_time_c / bin_size_c)
+
+        bins = np.arange(-max_time_c, max_time_c+bin_size_c, bin_size_c) * 1e3 / sampling_f
+        auto_corr = np.zeros(n_bins, dtype=np.int64)
+
+        for i in range(len(spike_train)):
+            for j in range(i+1, len(spike_train)):
```

Comment:
It is more or less the brut force solution for the problem.
The implementation I had in mind but more complicate is to increase `i` when the diff is becomming to big.
So you don't need a full `for i in range(n):` in short having a while loop that increase both i and j could much more efficient for the algo.

Reply:
> is to increase i when the diff is becomming to big
I believe this is what the `if diff > max_time: break` is doing.

> So you don't need a full `for i in range(n):`
I don't see how I can avoid this. Anyway you want to do this there is at least one substraction somewhere.

I'm not seeing how to optimize this further, can you elaborate.

Reply:
yes sorry. I miss this